elyra/kfp/operator.py

```diff
@@ -255,8 +263,15 @@ def __init__(
             self.container.set_memory_request(memory=str(mem_request) + "G")
 
         if self.gpu_limit:
-            gpu_vendor = self.pipeline_envs.get("GPU_VENDOR", "nvidia")
-            self.container.set_gpu_limit(gpu=str(gpu_limit), vendor=gpu_vendor)
+            if not self.gpu_vendor:
+                gpu_vendor = self.pipeline_envs.get("GPU_VENDOR", "nvidia")
+                self.container.set_gpu_limit(gpu=str(gpu_limit), vendor=gpu_vendor)
```

Comment:
Although `gpu_vendor` was not a member variable previously, we should set the member.  Also, for consistency, we should reference `gpu_limit` via `self`.
```suggestion
                self.gpu_vendor = self.pipeline_envs.get("GPU_VENDOR", "nvidia")
                self.container.set_gpu_limit(gpu=str(self.gpu_limit), vendor=self.gpu_vendor)
```